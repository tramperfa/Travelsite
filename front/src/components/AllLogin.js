import React from 'react';
import Button from 'material-ui/Button';
import {Link} from 'react-router-dom';

export default function AllLogin(props) {

  return (
    <div>
      <Button className="login" color="contrast" onClick={props.handleClickOpen}>Login</Button>
      <Link to="/signup">
        <Button className="signup" color="contrast">Signup</Button>
      </Link>
    </div>

  )
}
