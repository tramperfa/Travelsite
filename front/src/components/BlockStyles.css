.topContainter {
  min-width: 1080px;
}

.storyInfoBar {
  height: 80px;
  background-color: rgba(255,255,255,0.95);
  border-bottom: #d6d6d6 1px solid;
  box-shadow: 0 2px 5px rgba(0,0,0,0.12);
}
.storyContainer, .storyInfoContainer {
  width: 1000px;
  margin: 0 auto 0 auto;
}


.storyWrapper{
  display: flex;
  flex-direction: row;
}

.storyReader, .storyEditor{
  padding-top: 16px;
}

.storyEditor{
  cursor: text;
  text-align: left;
}

.sidebarWrapper{
  margin-left: 100px;
  padding: 16px;
  display: flex;
  align-self: flex-start;
  flex-direction: column;
  align-items: flex-start;
  position: sticky;
  top: 0;
  z-index: 10
}

.myAtomicStyle{
  width: 100%;
  margin-left: 0px;
  margin-right: 0px;
}

.atomicReader{
  margin-top: 16px;
  margin-bottom: 16px;
}

.atomicActionGroup{
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
}

.atomicActionItem{
  background-color: #FFFFFF;
  opacity: 0.75;
  z-index: 10;
  cursor: pointer;
}

.h1 {
  margin: 0px;
}

.subTitleContent{
  display: flex;
  flex-direction: row;
  background-color: #e4f0f5
}

.emojiSelect {
  position: absolute;
  cursor: pointer;
  z-index: 10;
  background-color: #ffffff
}

.emojiGroupTitle {
  margin-top: 8px;
  margin-bottom: 8px;
  color: #cccccc
}

.emojiGroup {
  display: flex;
  flex-flow: row wrap;
}

.emojiNavBar {
  padding-top: 8px;
  padding-bottom: 8px;
  margin-right: 12px;
  display: flex;
  flex-direction: row;
}

.navEntryActive .navEntry .emoji {
  width: 32px;
  height: 32px
}

.navEntryActive {
  opacity: 1.0
}

.navEntry {
  opacity: 0.75
}
.navEntry:hover {
  opacity: 1.0;
  background-color: #e1e1ea
}

.emoji:hover {
  background-color: #e1e1ea
}

.emojiImg {
  width: 28px;
  height: 28px
}

.sequenceAdjustor {
  cursor: pointer;
  height: 160px;
  line-height: 160px;
  margin-top: -80px;
  vertical-align: middle;
  position: fixed;
  background-color: rgba(129,130,131,0.95);
  color: #fff;
  top: 50%;
  right: 0
}

.headlineCropperCord {
  width: 100px;
  height: 30px;
  margin-left: -50px;
  margin-top: -15px;
  z-index: 10;
  background: rgba(255,255,255,0.4);
  border-radius: 5px;
  text-align: center;
  vertical-align: middle;
  line-height: 30px;
  font-size: 14px;
  text-shadow: 1px 1px 1px rgba(0,0,0,0.75);
  color: #fff
}

.headlineAction {
  display: flex;
  flex-direction: row;
  position: absolute;
  bottom: 0;
  right: 0;
}

.dialogActionItem{
  margin-left: 4px;
  background: rgba(255,255,255,0.4);
  border-radius: 5px;
}

.headlineWithoutImage {
  position: absolute;
  top: 20%;
  left: 10%
}

.headlineWithImage {
  background: rgba(255,255,255,0.4);
  padding-left: 6px;
  border-radius: 5px;
  position: absolute;
  bottom: 120px;
  left: 50%;
  margin-left: -500px;
}

.headlineContainer {
  position: relative;
}

.storyTitle {
  background: rgba(255,255,255,0.4);
  border-radius: 5px;
  padding: 6px;
  z-index: 10;
  position: absolute;
  width: 1000px;
  margin-left: -500px;
  left: 50%;
  bottom: 25px;
}

.storyReaderTitle {
  position: absolute;
  bottom: 0;
  left: 50%;
  width: 850px;
  margin-left: -500px;
  padding-left: 150px;
  color: #fff
}

.storyInfo{
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  display: flex;
  justify-content: space-between;
}

.authorInfo{
  padding: 10px 0 0 150px;
}

.storyAuthorAvatar {
  width: 124px;
  height: 124px;
  border-radius: 124px;
  position: relative;
  top: -60px;
  z-index: 10;
  cursor: pointer;
}

.readerActions{
  display: flex;
  flex-direction: row;
}

.readerActionItem{
  padding: 0 16px 0 16px;
  display: flex;
  flex-direction: column;
}

.paragraphGroupsWrapper {
  display: flex;
}

.paragraphWrapper {
  margin: 16px;
  width: 84px;
  display: flex;
  flex-direction: column;
  background-color: lightgreen;
}

.paragraphGroupsHeader {
  user-select: none;
}

.blockListWrapper {
  background-color: lightblue;
}

.sequenceDialog {
  width: 100%;
  height: 100%
}

.blockList {
  min-height: 300px;
}

.dndItem {
  width: 80px;
  align-items: center;
  border: 2px solid #bbb;
}

.dndImg {
  width: 100%;
  height: 100%
}

.textItemWrapper{
  position: relative;
}

.textItemContent {
  position: absolute;
  width: 360px;
  height: 150px;
  top: -2px;
  left: 80px;
  border: 2px solid #bbb;
  background: pink;
  z-index: 10;
  overflow: scroll;
  visibility: hidden;
}

.textItemWrapper:hover .textItemContent {
  visibility: visible;
}

.scrollTools {
  position: fixed;
  right: 50%;
  margin-right: -560px;
  bottom: 20px;
  display: flex;
  flex-direction: column;
  z-index: 20;
}

.scrollToolButton {
  border: 1px solid #e5e5e5;
  background-color: #fff;
}

.publishButton {
  width: max-content;
  margin: 50px auto 15px auto;
}
