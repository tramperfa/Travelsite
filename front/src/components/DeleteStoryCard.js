import React from 'react';
import PropTypes from 'prop-types';
import {graphql} from 'react-apollo';
import {withStyles} from 'material-ui/styles';
import Card, {CardContent, CardMedia} from 'material-ui/Card';
import IconButton from 'material-ui/IconButton';
import Typography from 'material-ui/Typography';
import Undo from "material-ui-icons/Undo";

import {RecoverStoryMutation} from '../graphql/story';
import {myDeleteStoryQuery} from '../graphql/story';

import imageTest from '../images/g.jpeg';

const styles = theme => ({
  card: {
    display: 'flex',
    marginTop: 24,
    width: 640,
    height: 300
  },
  details: {
    display: 'flex',
    flexDirection: 'column'
  },
  content: {
    flex: '1 0 auto'
  },
  cover: {
    width: 440,
    height: 300
  },
  textField: {
    textDecoration: 'none'
  }
});

function StoryCard(props) {
  const {classes, story} = props;

  const handleRecover = () => {
    props.recoverStory(story._id)
    console.log("TBD Recover");
  }

  return (
    <div>
      <Card className={classes.card}>
        <CardMedia className={classes.cover} image={imageTest}/>
        <div className={classes.details}>
          <CardContent className={classes.content}>
            <Typography type="headline">
              {story.title}
            </Typography>
            <div>{story.viewCount}
              Views
            </div>
            <div>
              {story.likeStoryCount}
              Likes
            </div>
            <IconButton aria-label="Undo" onClick={handleRecover}>
              Recovery Story
              <Undo/>
            </IconButton>
          </CardContent>
        </div>
      </Card>
    </div>
  );
}

StoryCard.propTypes = {
  classes: PropTypes.object.isRequired,
  //theme: PropTypes.object.isRequired,
  recoverStory: PropTypes.func.isRequired,
  story: PropTypes.object.isRequired
};

//NO REFETCH NEEDED
export const WithRecover = graphql(RecoverStoryMutation, {
  props: ({mutate}) => ({
    recoverStory: (storyID) => mutate({
      variables: {
        storyID: storyID
      },
      refetchQueries: [
        {
          query: myDeleteStoryQuery
        }
      ]
    })
  })
})

export default WithRecover(withStyles(styles, {withTheme: true})(StoryCard))
