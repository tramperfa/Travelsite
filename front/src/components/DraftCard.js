import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';
import {withStyles} from 'material-ui/styles';
import Card, {CardContent, CardMedia} from 'material-ui/Card';
import IconButton from 'material-ui/IconButton';
import Typography from 'material-ui/Typography';
import moment from 'moment';
import {graphql} from 'react-apollo';
import Edit from "material-ui-icons/Edit";
import Delete from "material-ui-icons/Delete";

//
import {draftsListQuery} from '../graphql/draft';
import {DeleteDraftMutation} from '../graphql/draft';

//
import imageTest from '../images/b.jpg';

const styles = theme => ({
  card: {
    display: 'flex',
    marginTop: 24,
    width: 620,
    height: 150
  },
  details: {
    display: 'flex',
    flexDirection: 'column'
  },
  content: {
    flex: '1 0 auto'
  },
  cover: {
    width: 220,
    height: 150
  },
  textField: {
    textDecoration: 'none'
  }
});

function DraftCard(props) {

  // class DraftCard extends React.Component {

  const handleDelete = () => {
    try {
      props.deleteDraft(props.draft._id)
    } catch (e) {
      console.log(e.graphQLErrors[0].message);
    } finally {}
  }

  // render() {
  const {classes, draft} = props;
  return (

    <div >
      <Card className={classes.card}>
        <Link className={classes.textField} to={`/edit/${draft._id}`}>
          <CardMedia
            className={classes.cover}
            image={imageTest}
            title="Live from space album cover"/>
        </Link>
        <div className={classes.details}>
          <Link className={classes.textField} to={`/edit/${draft._id}`}>
            <CardContent className={classes.content}>
              <Typography type="headline">
                {draft.title}
              </Typography>
              <Typography type="subheading" color="secondary">
                <div>
                  Last Updated: {moment(new Date(draft.lastUpdate)).utc().local().format("YYYY-MM-DD HH:mm")}
                </div>
              </Typography>
            </CardContent>
          </Link>
          <div className={classes.controls}>
            Continue Writing
            <IconButton aria-label="Edit">
              <Edit/>
            </IconButton>

            <IconButton aria-label="Delete" onClick={handleDelete}>
              <Delete/>
            </IconButton>

          </div>
        </div>

      </Card>
    </div>
  );
  // }
}

DraftCard.propTypes = {
  classes: PropTypes.object.isRequired,
  //theme: PropTypes.object.isRequired,
  deleteDraft: PropTypes.func.isRequired,
  draft: PropTypes.object.isRequired
};

export const WithDelete = graphql(DeleteDraftMutation, {
  props: ({mutate}) => ({
    deleteDraft: (draftID) => mutate({
      variables: {
        draftID: draftID
      },
      refetchQueries: [
        {
          query: draftsListQuery
        }
      ]
    })
  })
})

export default WithDelete(withStyles(styles, {withTheme: true})(DraftCard))
