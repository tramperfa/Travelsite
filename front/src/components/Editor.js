import React from 'react';

const Editor = ({ match }) => {
  return (
    <div className="NotFound">Place Holder for Editor</div>
  );
};

export default Editor;
