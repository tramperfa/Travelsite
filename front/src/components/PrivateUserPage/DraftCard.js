import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';
import format from 'date-fns/format'
import {compose} from 'recompose';

//
import {withStyles} from 'material-ui/styles';
import Card, {CardContent, CardMedia} from 'material-ui/Card';
import IconButton from 'material-ui/IconButton';
import Typography from 'material-ui/Typography';
import Edit from "material-ui-icons/Edit";
import Delete from "material-ui-icons/Delete";

//
import {WithDeleteDraftMutation} from '../../graphql/draft';
import ErrorComp from '../../lib/ErrorComp';
import {error, onError} from '../../lib/utils';
import CONSTS from '../../lib/consts';

//
import defaultbrowserCoverImage from '../../images/browserCoverImage.png';

export const styles = theme => ({
	card: {
		display: 'flex',
		marginTop: 24,
		width: 620,
		height: 150
	},
	details: {
		display: 'flex',
		flexDirection: 'column'
	},
	content: {
		flex: '1 0 auto'
	},
	cover: {
		width: 220,
		height: 150
	},
	textField: {
		textDecoration: 'none'
	}
});

class DraftCard extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			error: error
		}
	}

	handleDelete = () => {
		this.props.deleteDraft(this.props.draft._id).then(() => {
			//this.setState({deleteRedirect: true})
		}).catch((err) => {
			this.setState(onError(err))
		})
	}

	render() {
		const {classes, draft} = this.props;
		return (
			<div>
				<ErrorComp error={this.state.error}/>
				<Card className={classes.card}>
					<Link className={classes.textField} to={`/my/edit/${draft._id}`}>
						<CardMedia
							className={classes.cover}
							image={draft.coverImage
								? CONSTS.BUCKET_NAME + draft.coverImage.browserCoverImage.filename
								: defaultbrowserCoverImage}/>
					</Link>
					<div className={classes.details}>
						<Link className={classes.textField} to={`/my/edit/${draft._id}`}>
							<CardContent className={classes.content}>
								<Typography type="headline">
									{draft.title}
								</Typography>
								<Typography type="subheading" color="secondary">
									<div>
										Last Updated: {format(new Date(draft.lastUpdate), "YYYY-MM-DD HH:mm")}
									</div>
								</Typography>
							</CardContent>
						</Link>
						<div className={classes.controls}>
							Continue Writing
							<IconButton aria-label="Edit">
								<Edit/>
							</IconButton>
							<IconButton aria-label="Delete" onClick={this.handleDelete}>
								<Delete/>
							</IconButton>
						</div>
					</div>
				</Card>
			</div>
		);

	}

}

DraftCard.propTypes = {
	classes: PropTypes.object.isRequired,
	deleteDraft: PropTypes.func.isRequired,
	draft: PropTypes.object.isRequired
};

export default compose(WithDeleteDraftMutation, withStyles(styles))(DraftCard)
