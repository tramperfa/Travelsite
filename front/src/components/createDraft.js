import React from 'react';
import { gql, graphql } from 'react-apollo';



const createStory = ({ mutate }) => {
  const handleKeyUp = (evt) => {
    if (evt.keyCode === 13) {
      mutate({
        variables: { user_id: '1'},
      });
      evt.target.value = '';
    }
  };

  return (
    evt.target.value()
    <input
      type="text"
      placeholder="New story"
      onKeyUp={handleKeyUp}
    />
  );
};

const createStoryMutation = gql`
  mutation createStory($user_id: String!) {
    createStory(user_id: $user_id) {
      _id
      title
      user_id
    }
  }
`;


const createStoryWithMutation = graphql(
  createStoryMutation,
)(createStory);

export default createStoryWithMutation;
