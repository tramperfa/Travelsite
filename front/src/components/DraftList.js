import React from 'react';
import {graphql} from 'react-apollo';

//
import {DRAFTS_LIST_QUERY} from '../graphql/draft';

//
import DraftCard from './DraftCard';

const draftsList = ({
	data: {
		loading,
		error,
		myDrafts
	}
}) => {
	if (loading) {
		return <p>Loading ...</p>;
	}
	if (error) {
		return <p>{error.message}</p>;
	}
	//console.log(myDrafts);

	return (
		<div>
			{
				myDrafts.map(draft => (
					<div key={draft._id} className='draft'>
						<DraftCard draft={draft}/>
					</div>
				))
			}
		</div>
	);
};

export default graphql(DRAFTS_LIST_QUERY, {
	options: {
		fetchPolicy: 'network-only'
	}
})(draftsList);
