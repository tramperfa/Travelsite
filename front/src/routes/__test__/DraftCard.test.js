import React from 'react';
// Note: test renderer must be required after react-native. import renderer from
// 'react-test-renderer';
import {DraftCard} from '../../components/DraftCard';
import {shallow} from 'enzyme';

test('renders correctly', () => {

	const wrapper = shallow(
		<DraftCard deleteDraft={() => {}} draft={{}} classes={{}}/>,

	);
	expect(wrapper).toMatchSnapshot();
	// const myButton = wrapper.find('withStyles(Button)'); console.log(myButton);
	// wrapper.find('button').simulate('click');

});
