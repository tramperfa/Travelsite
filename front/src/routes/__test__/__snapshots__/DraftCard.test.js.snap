// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classes={Object {}}
    deleteDraft={[Function]}
    draft={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Card
        className={undefined}
        raised={false}
>
        <Link
                className={undefined}
                replace={false}
                to="/edit/undefined"
        >
                <withStyles(CardMedia)
                        className={undefined}
                        image="b.jpg"
                        title="Live from space album cover"
                />
        </Link>
        <div
                className={undefined}
        >
                <Link
                        className={undefined}
                        replace={false}
                        to="/edit/undefined"
                >
                        <withStyles(CardContent)
                                className={undefined}
                        >
                                <withStyles(Typography)
                                        type="headline"
                                />
                                <withStyles(Typography)
                                        color="secondary"
                                        type="subheading"
                                >
                                        <div>
                                                Last Updated: 
                                                Invalid date
                                        </div>
                                </withStyles(Typography)>
                        </withStyles(CardContent)>
                </Link>
                <div
                        className={undefined}
                >
                        Continue Writing
                        <withStyles(IconButton)
                                aria-label="Edit"
                        >
                                <pure(Edit) />
                        </withStyles(IconButton)>
                        <withStyles(IconButton)
                                aria-label="Delete"
                                onClick={[Function]}
                        >
                                <pure(Delete) />
                        </withStyles(IconButton)>
                </div>
        </div>
</Card>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Link
            className={undefined}
            replace={false}
            to="/edit/undefined"
>
            <withStyles(CardMedia)
                        className={undefined}
                        image="b.jpg"
                        title="Live from space album cover"
            />
</Link>,
          <div
            className={undefined}
>
            <Link
                        className={undefined}
                        replace={false}
                        to="/edit/undefined"
            >
                        <withStyles(CardContent)
                                    className={undefined}
                        >
                                    <withStyles(Typography)
                                                type="headline"
                                    />
                                    <withStyles(Typography)
                                                color="secondary"
                                                type="subheading"
                                    >
                                                <div>
                                                            Last Updated: 
                                                            Invalid date
                                                </div>
                                    </withStyles(Typography)>
                        </withStyles(CardContent)>
            </Link>
            <div
                        className={undefined}
            >
                        Continue Writing
                        <withStyles(IconButton)
                                    aria-label="Edit"
                        >
                                    <pure(Edit) />
                        </withStyles(IconButton)>
                        <withStyles(IconButton)
                                    aria-label="Delete"
                                    onClick={[Function]}
                        >
                                    <pure(Delete) />
                        </withStyles(IconButton)>
            </div>
</div>,
        ],
        "className": undefined,
        "raised": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <withStyles(CardMedia)
              className={undefined}
              image="b.jpg"
              title="Live from space album cover"
/>,
            "className": undefined,
            "replace": false,
            "to": "/edit/undefined",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": undefined,
              "image": "b.jpg",
              "title": "Live from space album cover",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className={undefined}
                replace={false}
                to="/edit/undefined"
>
                <withStyles(CardContent)
                                className={undefined}
                >
                                <withStyles(Typography)
                                                type="headline"
                                />
                                <withStyles(Typography)
                                                color="secondary"
                                                type="subheading"
                                >
                                                <div>
                                                                Last Updated: 
                                                                Invalid date
                                                </div>
                                </withStyles(Typography)>
                </withStyles(CardContent)>
</Link>,
              <div
                className={undefined}
>
                Continue Writing
                <withStyles(IconButton)
                                aria-label="Edit"
                >
                                <pure(Edit) />
                </withStyles(IconButton)>
                <withStyles(IconButton)
                                aria-label="Delete"
                                onClick={[Function]}
                >
                                <pure(Delete) />
                </withStyles(IconButton)>
</div>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <withStyles(CardContent)
                  className={undefined}
>
                  <withStyles(Typography)
                                    type="headline"
                  />
                  <withStyles(Typography)
                                    color="secondary"
                                    type="subheading"
                  >
                                    <div>
                                                      Last Updated: 
                                                      Invalid date
                                    </div>
                  </withStyles(Typography)>
</withStyles(CardContent)>,
                "className": undefined,
                "replace": false,
                "to": "/edit/undefined",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <withStyles(Typography)
                      type="headline"
/>,
                    <withStyles(Typography)
                      color="secondary"
                      type="subheading"
>
                      <div>
                                            Last Updated: 
                                            Invalid date
                      </div>
</withStyles(Typography)>,
                  ],
                  "className": undefined,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": undefined,
                      "type": "headline",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <div>
                        Last Updated: 
                        Invalid date
</div>,
                      "color": "secondary",
                      "type": "subheading",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "Last Updated: ",
                          "Invalid date",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        "Last Updated: ",
                        "Invalid date",
                      ],
                      "type": "div",
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Continue Writing",
                  <withStyles(IconButton)
                    aria-label="Edit"
>
                    <pure(Edit) />
</withStyles(IconButton)>,
                  <withStyles(IconButton)
                    aria-label="Delete"
                    onClick={[Function]}
>
                    <pure(Delete) />
</withStyles(IconButton)>,
                ],
                "className": undefined,
              },
              "ref": null,
              "rendered": Array [
                "Continue Writing",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "aria-label": "Edit",
                    "children": <pure(Edit) />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "aria-label": "Delete",
                    "children": <pure(Delete) />,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Card
          className={undefined}
          raised={false}
>
          <Link
                    className={undefined}
                    replace={false}
                    to="/edit/undefined"
          >
                    <withStyles(CardMedia)
                              className={undefined}
                              image="b.jpg"
                              title="Live from space album cover"
                    />
          </Link>
          <div
                    className={undefined}
          >
                    <Link
                              className={undefined}
                              replace={false}
                              to="/edit/undefined"
                    >
                              <withStyles(CardContent)
                                        className={undefined}
                              >
                                        <withStyles(Typography)
                                                  type="headline"
                                        />
                                        <withStyles(Typography)
                                                  color="secondary"
                                                  type="subheading"
                                        >
                                                  <div>
                                                            Last Updated: 
                                                            Invalid date
                                                  </div>
                                        </withStyles(Typography)>
                              </withStyles(CardContent)>
                    </Link>
                    <div
                              className={undefined}
                    >
                              Continue Writing
                              <withStyles(IconButton)
                                        aria-label="Edit"
                              >
                                        <pure(Edit) />
                              </withStyles(IconButton)>
                              <withStyles(IconButton)
                                        aria-label="Delete"
                                        onClick={[Function]}
                              >
                                        <pure(Delete) />
                              </withStyles(IconButton)>
                    </div>
          </div>
</Card>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Link
              className={undefined}
              replace={false}
              to="/edit/undefined"
>
              <withStyles(CardMedia)
                            className={undefined}
                            image="b.jpg"
                            title="Live from space album cover"
              />
</Link>,
            <div
              className={undefined}
>
              <Link
                            className={undefined}
                            replace={false}
                            to="/edit/undefined"
              >
                            <withStyles(CardContent)
                                          className={undefined}
                            >
                                          <withStyles(Typography)
                                                        type="headline"
                                          />
                                          <withStyles(Typography)
                                                        color="secondary"
                                                        type="subheading"
                                          >
                                                        <div>
                                                                      Last Updated: 
                                                                      Invalid date
                                                        </div>
                                          </withStyles(Typography)>
                            </withStyles(CardContent)>
              </Link>
              <div
                            className={undefined}
              >
                            Continue Writing
                            <withStyles(IconButton)
                                          aria-label="Edit"
                            >
                                          <pure(Edit) />
                            </withStyles(IconButton)>
                            <withStyles(IconButton)
                                          aria-label="Delete"
                                          onClick={[Function]}
                            >
                                          <pure(Delete) />
                            </withStyles(IconButton)>
              </div>
</div>,
          ],
          "className": undefined,
          "raised": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <withStyles(CardMedia)
                className={undefined}
                image="b.jpg"
                title="Live from space album cover"
/>,
              "className": undefined,
              "replace": false,
              "to": "/edit/undefined",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": undefined,
                "image": "b.jpg",
                "title": "Live from space album cover",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  className={undefined}
                  replace={false}
                  to="/edit/undefined"
>
                  <withStyles(CardContent)
                                    className={undefined}
                  >
                                    <withStyles(Typography)
                                                      type="headline"
                                    />
                                    <withStyles(Typography)
                                                      color="secondary"
                                                      type="subheading"
                                    >
                                                      <div>
                                                                        Last Updated: 
                                                                        Invalid date
                                                      </div>
                                    </withStyles(Typography)>
                  </withStyles(CardContent)>
</Link>,
                <div
                  className={undefined}
>
                  Continue Writing
                  <withStyles(IconButton)
                                    aria-label="Edit"
                  >
                                    <pure(Edit) />
                  </withStyles(IconButton)>
                  <withStyles(IconButton)
                                    aria-label="Delete"
                                    onClick={[Function]}
                  >
                                    <pure(Delete) />
                  </withStyles(IconButton)>
</div>,
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <withStyles(CardContent)
                    className={undefined}
>
                    <withStyles(Typography)
                                        type="headline"
                    />
                    <withStyles(Typography)
                                        color="secondary"
                                        type="subheading"
                    >
                                        <div>
                                                            Last Updated: 
                                                            Invalid date
                                        </div>
                    </withStyles(Typography)>
</withStyles(CardContent)>,
                  "className": undefined,
                  "replace": false,
                  "to": "/edit/undefined",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <withStyles(Typography)
                        type="headline"
/>,
                      <withStyles(Typography)
                        color="secondary"
                        type="subheading"
>
                        <div>
                                                Last Updated: 
                                                Invalid date
                        </div>
</withStyles(Typography)>,
                    ],
                    "className": undefined,
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": undefined,
                        "type": "headline",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": <div>
                          Last Updated: 
                          Invalid date
</div>,
                        "color": "secondary",
                        "type": "subheading",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            "Last Updated: ",
                            "Invalid date",
                          ],
                        },
                        "ref": null,
                        "rendered": Array [
                          "Last Updated: ",
                          "Invalid date",
                        ],
                        "type": "div",
                      },
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Continue Writing",
                    <withStyles(IconButton)
                      aria-label="Edit"
>
                      <pure(Edit) />
</withStyles(IconButton)>,
                    <withStyles(IconButton)
                      aria-label="Delete"
                      onClick={[Function]}
>
                      <pure(Delete) />
</withStyles(IconButton)>,
                  ],
                  "className": undefined,
                },
                "ref": null,
                "rendered": Array [
                  "Continue Writing",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "aria-label": "Edit",
                      "children": <pure(Edit) />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "aria-label": "Delete",
                      "children": <pure(Delete) />,
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
